### PR TITLE
Adding mirror option to git module

### PR DESCRIPTION
**Pull request by mhumeSF**
##### SUMMARY
Adding `--mirror` option to git module

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
- Git

##### ANSIBLE VERSION
```
ansible 2.3.0.0
  config file = 
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Apr  4 2017, 08:47:57) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]
```

### CODE REVIEW

**Comment by mhumeSF** [comment]
I should add some logic to throw a warning if `bare` is used with `mirror` in tandem

File: lib/ansible/modules/source_control/git.py [hunk 1]
```diff
@@ -482,7 +490,7 @@ def clone(git_path, module, repo, dest, remote, depth, version, bare,
         verify_commit_sign(git_path, module, dest, version)
 
 
-def has_local_mods(module, git_path, dest, bare):
+def has_local_mods(module, git_path, dest, bare, mirror):
```
Comment (robinro) [inline lib/ansible/modules/source_control/git.py]: Why is the new argument needed here?

**Review by robinro — CHANGES_REQUESTED** [review]
@mhumeSF Thanks for this PR.

Have you looked into idempotency of the mirror operation?
How does it behave when some ref/... changes on the remote?

https://stackoverflow.com/questions/6150188/how-to-update-a-git-clone-mirror
sound like `fetch` is not enough to update a `mirror` remote and some additional logic might be needed.

Maybe in general: The new argument is added to quite a few places and in the long term we should refactor this to only pass an `args` variable around or reuse the `module` variable. But this should be a separate change that someone else or myself can tackle.

**Comment by mhumeSF** [comment]
I will have to give more thought on idempotency of the mirror operation. I do agree the `bare` and `mirror` flags could probably be better refactored into some sorta `args` var.

**Pull request closed without merging**